src/coreclr/jit/importercalls.cpp

```diff
@@ -32,6 +32,107 @@
 #pragma warning(disable : 21000) // Suppress PREFast warning about overly large function
 #endif
 
+bool Compiler::impTryFindField(CORINFO_METHOD_HANDLE methHnd, CORINFO_RESOLVED_TOKEN* pResolvedToken, OPCODE* opcode)
+{
```

Comment:
This should issue `beginInlining` / `reportInliningDecision` callbacks.

Reply:
Like this?

Reply:
This seems like we can end up issuing duplicate callbacks now (e.g. this one failing, then regular inlining succeeding). In general it seems unfortunate to end up with two separate inliners if we can avoid it, so some way of integrating this behavior into the existing inliner would be much preferred. For example, this new inlining should most likely affect the inline tree that gets reported by `reportRichMappings` too, and proper debug information should be created to refer to it.